contrib/sarplus/scala/build.sbt

```diff
@@ -1,30 +1,103 @@
+    "org.apache.spark" %% "spark-core" % sparkVer.value,
+    "org.apache.spark" %% "spark-mllib" % sparkVer.value,
+    "org.apache.spark" %% "spark-sql" % sparkVer.value,
+    "org.scala-lang" % "scala-reflect" % scalaVersion.value,
+    "com.google.guava" % "guava" % "15.0",
+    "org.scalamock" %% "scalamock" % "4.1.0" % "test",
+    "org.scalatest" %% "scalatest" % "3.0.8" % "test",
+    "xerces" % "xercesImpl" % "2.12.1",
+  ),
+  Compile / packageBin / artifact := {
+    val prev: Artifact = (Compile / packageBin / artifact).value
+    prev.withClassifier(
+        prev.classifier match {
+          case None => {
+            val splitVer = sparkVer.value.split('.')
+            val major = splitVer(0).toInt
+            val minor = splitVer(1).toInt
+            if (major >=3 && minor >= 2) Some("spark32") else None
+          }
+          case Some(s: String) => Some(s)
+        }
+    )
+  },
+)
+
+lazy val compat = project.settings(commonSettings)
+lazy val root = (project in file("."))
+  .dependsOn(compat)
+  .settings(
+    name := "sarplus",
+    commonSettings,
+  )
 
-version := "0.2.6" 
 
-sparkComponents ++= Seq("core", "sql", "mllib")
+// POM metadata configuration.  See https://www.scala-sbt.org/release/docs/Using-Sonatype.html
 
-libraryDependencies ++= Seq(
-  "commons-io" % "commons-io" % "2.6",
-  "com.google.guava" % "guava" % "25.0-jre",
-  "org.scalatest" %% "scalatest" % "3.0.5" % "test",
-  "org.scalamock" %% "scalamock" % "4.1.0" % "test"
+organization := "com.microsoft.sarplus"
+organizationName := "microsoft"
+organizationHomepage := Some(url("https://microsoft.com"))
+
+scmInfo := Some(
+  ScmInfo(
+    url("https://github.com/microsoft/recommenders/tree/main/contrib/sarplus"),
+    "scm:git@github.com:microsoft/recommenders.git"
+  )
+)
+
+developers := List(
+  Developer(
+    id = "eisber",
+    name = "Markus Cozowicz",
+    email = "marcozo@microsoft.com",
+    url = url("https://github.com/eisber")
+  )
 )
 
-// All Spark Packages need a license
+description := "sarplus"
 licenses := Seq("MIT" -> url("http://opensource.org/licenses/MIT"))
+homepage := Some(url("https://github.com/microsoft/recommenders/tree/main/contrib/sarplus"))
+pomIncludeRepository := { _ => false }
+publishTo := {
+  val nexus = "https://oss.sonatype.org/"
+  if (isSnapshot.value) Some("snapshots" at nexus + "content/repositories/snapshots")
+  else Some("releases" at nexus + "service/local/staging/deploy/maven2")
+}
+publishMavenStyle := true
 
-// doesn't work anyway...
-credentials += Credentials(Path.userHome / ".ivy2" / ".sbtcredentials") // A file containing credentials
 
-spHomepage := "http://github.com/Microsoft/Recommenders/contrib/sarplus"
+// PGP key configuration
 
-// If you published your package to Maven Central for this release (must be done prior to spPublish)
-spIncludeMaven := true
+credentials += Credentials(
+  "GnuPG Key ID",
+  "gpg",
+  "C72E596B384EC14CFA65D80A36CB250AF1C18ECE",
+  "ignored"
+)

```

Comment:
this can be problematic, we are getting a lot of automatic messages because we are adding tokens in the code. Can we remove this and add it in the pipeline somehow?

Reply:
Removed.